### DISCUSSION TITLE
Extend results to more types

### DISCUSSION THREAD

**Discussion by CyberFox001** [alphapapa/org-ql#452] [Ideas]
### OS/platform

GNU/Linux

### Emacs version and provenance

Emacs 29.4, from Fedora repository

### Org version and provenance

Org-mode 9.7.9, from GNU ELPA

### org-ql package version and provenance

Org-ql 20240403.2027, from MELPA

### Description

Today, Org-ql let user search for entries into multiple Org-mode files. Each entries is an Org-mode heading, that could be a note, a task or a document section.

This feature request is about extend the result to more types of results, like:
- Org-mode file
- Images
- Bloc of code
- Tables
- Or any element that could be named
- Links

This would let user search for other elements in files, like the schema they have done on another note with a PlantUML code block, or the image they have added to this other task. They could also search for Org-mode files, based on `#+TITLE`, `#+AUTHOR` or `#+DATE` values.

This feature could open possibilities for new features like:
- With a dynamic block, built a table of each Org-mode files present in a folder, sort them by date and show their author, date and their title as a link to the file (Can be useful to make the home page of a blog made with Org-mode)
- Copy an image or source code bloc from another note to the current one

Such feature is a big change, and it's maybe out of scope for Org-ql.

**Comment by alphapapa** [comment]
Hi,

These are interesting ideas, some of which have been considered and discussed before.  In the long-term, I'm open to the possibility of expanding the scope of Org QL to do more things like this (e.g. searching lists, logbook entries, etc).  In the short-term, some of those things are possible already since a query can call any function to return any kind of result desired.  For example, see the implementation of the command `org-ql-open-link`.

Other than that, feel free to brainstorm in a discussion here, and please be specific when possible.

**Reply by CyberFox001** [reply]
Thank you for your answer. 

I will take a look in `org-ql-open-link`.
